core/__tests__/tasks/profile/sweep.ts

```diff
@@ -178,4 +152,29 @@ describe("tasks/profiles:sweep", () => {
     await expect(_import.reload()).rejects.toThrow(/does not exist anymore/);
     await expect(_export.reload()).rejects.toThrow(/does not exist anymore/);
   });
+
+  // this test should be last
+  test("deletes profiles when there is no directlyMapped property", async () => {
+    const profile: Profile = await helper.factories.profile();
+    await profile.addOrUpdateProperties({
+      userId: [1000],
+      isVIP: [true],
+      ltv: [213],
+    });
+
+    await ProfileProperty.update(
+      { state: "ready" },
+      { where: { profileId: profile.id } }
+    );
+    await profile.update({ state: "ready" });
+    let count = await specHelper.runTask("profiles:sweep", {});
+    expect(count).toBe(0);
+
+    // now remove the directly mapped property
+    // @ts-ignore
```

Comment:
curious - why was this needed?

Reply:
It's because I wanted to skip the hooks - Sequelize really doesn't want you doing that.  If I had left the hooks on like normal, the @BeforeDestory checks would fire and I'd get yelled at because the mapping was still in use or something like that